### ISSUE TITLE
Git / Localization platform sync

### ISSUE THREAD

**Issue by phylophyl** [freedombox/Plinth#319]
1 - Transifex does not provide Git sync apparently, https://hosted.weblate.org/ does
2 - How to create a new langage in Plinth/local directory

Sub-tasks (added by @SunilMohanAdapa, please keep editing appropriately)
- [x] Decide which of each translation team move to Weblate:
  - [x] Danish (Weblate)
    - [x] Remove from Transifex
  - [x] Dutch (Weblate)
    - [x] Remove from Transifex
  - [x] French (Weblate)
    - [x] Remove from Transifex
  - [x] German (Weblate)
    - [x] Remove from Transifex
  - [x] Norwegian Bokmål  (Transifex)
  - [x] Telugu (Weblate)
    - [x] Remove from Transifex
- [x] Weblate
  - [X] Remove `dk` locale
  - [X] Address `fake` locale (it is only for developers, it is not a problem for syncing)
  - [x] Pull (by Weblate, from GitHub)
    - [X] Add GitHub hook service for Weblate
  - [x] Push (by Weblate, to GitHub)
    - [x] Add Weblate user as contributor with write access
    - [x] Notify Weblate team that we are interested in doing this. Do not do this yet.
- [x] Transifex
  - [x] Pull (by Transifex, from GitHub)
    - [X] Create POT file
  - [x] Push (by Transifex, to GitHub)

**Comment by SunilMohanAdapa** [comment]
On 11/30/2015 02:37 PM, Philippe Baret wrote:
> 1 - Transifex does not provide Git sync apparently,
> https://hosted.weblate.org/ does

We could perhaps setup FreedomBox project there as well and move as many
teams as possible over there.  Transifex can help with people
discovering our project and weblate can be our preferred solution.

> 2 - How to create a new langage in Plinth/local directory

Simply create a new file with the proper directory name.  For example,
create a file called plinth/locale/da/LC_MESSAGES/django.po.  This file
will be automatically compiled and installed into the system when
installing Plinth.

-- 
Sunil

**Comment by phylophyl** [comment]
1 FreedomBox project setup in process at https://hosted.weblate.org/ (form filled and sent via my mail weblate@hyperlecture.net)

2 Danish (DA) path and django.po file entered

**Comment by phylophyl** [comment]
Tricks read on the discussion list to investigate :

-  This can be solved by storing the generated POT file in git and pointing
Transifex to this file.  Then Transifex will update the original texts
one per day. [Petter Reinholdtsen]

- It is possible to produce a generic .pot by doing "django-admin
makemessages --keep-pot [-a|-l <locale>]". This seems to me to be the
right way to do it. And then a Transifex maintainer should point
Transifex there as per your advice. [Mikkel Kirkgaard Nielsen]

**Comment by phylophyl** [comment]
Registration email from Weblate:

Dne 30.11.2015 v 17:34 noreply@weblate.org napsal(a):
>
> Philippe Baret <weblate@hyperlecture.net> wants to host FreedomBox
>
> Project:    FreedomBox
> Website:    https://wiki.debian.org/FreedomBox/
> Repository: https://github.com/freedombox/Plinth
> Filemask:   https://github.com/freedombox/Plinth/tree/master/plinth/locale
> Username:   phylophyl
>
> Additional message:
>
> Hello,
> FreedomBox is a Debian Pure Blend project.
> Best Regards,
> Philippe Baret

I've just imported it:

https://hosted.weblate.org/projects/freedombox/plinth/

Please check that everything looks correct. If you want Weblate to
automatically follow and push to your repository, please follow
instructions on http://weblate.org/en/hosting/, namely:

To notify Weblate about changes in your repository please add Weblate
Service to your project:

https://docs.weblate.org/en/latest/admin/continuous.html#github-setup

If you want Weblate to push changes directly to your repository, please
add weblate user to the repository:

https://docs.weblate.org/en/latest/admin/continuous.html#pushing-changes-from-hosted-weblate

PS: Few oddities I've noticed:

- dk language code does not exist
- fake translation also looks a bit weird

For now I've set Weblate to ignore fake one, but the dk shows there.
Please rename/remove these as appropriate.

-- 
	Michal Čihař | http://cihar.com | http://blog.cihar.com

**Comment by mikini** [comment]
-----BEGIN PGP SIGNED MESSAGE-----
Hash: SHA1

On 2015-12-02 11:47, Philippe Baret wrote:
> Registration email from Weblate:

Nice. Just registered as user, the translator's interface looks very
smooth and more lean than Transifex.

> * dk language code does not exist

As I pointed out in previous post
(http://lists.alioth.debian.org/pipermail/freedombox-discuss/2015-Decemb
er/007071.html)
I also think dk should be removed.

> * fake translation also looks a bit weird

Sunil added fake as a test according to commitlog
(https://github.com/freedombox/Plinth/commit/b84534eaec214194060687d42c5
92b6c9a821aab),
should be fine to ignore or remove completely if it is obsolete.
Django won't be able to show it anyway if Django hasn't got a
fake-locale defined too.

- -- 
    Mikkel

-----BEGIN PGP SIGNATURE-----
Version: GnuPG v2

iQEcBAEBAgAGBQJWXtWPAAoJEJ2luFWzaTSaz9wH/j2+QqDGHB3drrt6wsf9C/HZ
NFosy2BqvQfjZDitenLBUUhrBkk1weZD2B0oMRPBFJ6TK/q1J6CeJrIO+1qtuH5T
yWryFSMn1pDzrAUHSLACxmYLFcvanX1pFdicsyhPwxgP417s2gBtaUTqIOwA/jGS
IORqlEbYu0zDMZ/INmqbsEI9BDpAAF5+BqBXLxLHhRFxk92nL6HmnjR6hJtZpTxA
OFj39A20jg3LYNbvli3Ce7WuyoWAWivSgu+V8H8RPueDaYNrvpuF42zrSMg/s/vt
9+1OUc68qEy5zb3JMU0SHxuMwN9ycXIUWp8sGhh7tCaku5YaaCVWktXh8DC6iVo=
=PVUx
-----END PGP SIGNATURE-----

**Comment by phylophyl** [comment]
DK is removed

**Comment by SunilMohanAdapa** [comment]
On the first comment I have added subtasks required for this task.

**Comment by phylophyl** [comment]
Norwegian Bokmål will stay on Transifex, Pere team is not reacting yet, they are busy
Mikkel, do you want to move or stay ?

**Comment by mikini** [comment]
As previously stated on the ml I personally see no reason to keep the Danish plinth translation in Transifex. 

How is the setup on Weblate moving along? Are we syncing to git soon? If not it might be better to remove the old .po's present there. Somebody (saw some edits in de by ikmaak also, maybe it was him?) has been editing the outdated Danish strings present there according to its auto hints (whitespace and punctuation). Most of them fixes I have already done in current hit. That is a bit counterproductive as it will probably cause conflicts that has to be handled manually when merging/syncing.

**Comment by phylophyl** [comment]
> How is the setup on Weblate moving along?
Sunil is talking with Weblate guy to get simple commits (not merge commits) made during pushes.

**Comment by SunilMohanAdapa** [comment]
I have observed that Weblate is pulling the content from our Git. So, the hooks are working.

**Comment by SunilMohanAdapa** [comment]
Push from Weblate to GitHub seems to be working properly in the way wanted (by having rebased commits).

We need to delete the migrated teams from Transifex to avoid problems there.

**Comment by phylophyl** [comment]
All teams except NB can be deleted, is it a problem ?

**Comment by SunilMohanAdapa** [comment]
On 12/10/2015 12:25 AM, Philippe Baret wrote:
> All teams except NB can be deleted, is it a problem ?

I don't believe there should be problem technically.  I just took a
backup of the .PO files in their current state, just in case.

-- 
Sunil

**Comment by phylophyl** [comment]
I guess Push (by Transifex, to GitHub) is the only problem left ? There has been a solution published on the discussion list.

**Comment by SunilMohanAdapa** [comment]
On 12/14/2015 06:32 PM, Philippe Baret wrote:
> I guess Push (by Transifex, to GitHub) is the only problem left ? There
> has been a solution published on the discussion list.

I will check the solution.

We also need to remove many translations from Transifex to avoid
clashes, rework and confusion among translators.

-- 
Sunil

**Comment by phylophyl** [comment]
I tired to remove translations except norwegian, content is deleted but it is still impossible to withdraw the language, it is a problem if the content from github fill them again.

**Comment by SunilMohanAdapa** [comment]
On 12/19/2015 01:13 AM, Philippe Baret wrote:
> I tired to remove translations except norwegian, content is deleted but
> it is still impossible to withdraw the language, it is a problem if the
> content from github fill them again.

Thanks to Philippe, translations for groups other than Norwegian Bokmål
are now removed and there is an announcement made about contacting the
teams on IRC/mailing list.  I update the bug status accordingly.  The
last remaining item is for me to setup the pull from Transifex.

I also updated the project's long description mentioning the fact the
most language teams are using other tools and anyone looking to
translate should just raise a request and find out.

-- 
Sunil
